samples/azure/azure-service-fabric-routing/sample.md

```diff
@@ -0,0 +1,202 @@
+
+Example:
+
+- Vote for John; cast in 88701; should result in a `PlaceVote` message routed to named partition "John", followed by a `TrackZipCode` message routed to range partition (66000 -> 98999).
+- Vote for Abby; cast in 36458; should result in a `PlaceVote` message routed to named partition "Abby", followed by a `TrackZipCode` message routed to range partition (33000 -> 65999).
+- Vote for John; cast in 12789; should result in a `PlaceVote` message routed to named partition "John", followed by a `TrackZipCode` message routed to range partition (0 -> 32999).
+
+Partition-aware behavior is achieved by combining sender-side distribution (a built-in NServiceBus feature) with a few modifications to the message processing pipeline.
+
+The remainder of this document will focus on the different techniques that can be used to configure these distribution strategies, either manually or automatically, to achieve full partition aware routing. 
+
+
+## Partitioned Endpoint Configuration
+
+Endpoint instances hosted with stateful service replicas need to be uniquely addressable by the partition key associated with each replica. Possible keys are defined by the `NamedPartition` and `Int64RangePartition` info obtained from Service Fabric. The unique partition keys are used for `SendLocal` and `ReplyTo` operations to route messages to the correct partitions.
+
+Partitioned endpoints require specifying partitioning configuration by calling an extension method on `EndpointConfiguration`:
+
+snippet: ApplyPartitionConfigurationToEndpoint-ZipCodeVoteCount
+
+### Local sends
+
+All local sends are handled by the `PartitionAwareDistributionStrategy`. For each locally sent message it sets the `partition-key` header value to the local partition key and routes the message to the queue associated with the local partition.
+
+### Replies
+
+When replying, an endpoint routes the reply message to the endpoint that initiated the conversation. It's the responsibility of the requestor to properly set the reply-to header before sending out the request. For a partitioned endpoint this implies that it sets the reply to header to its instance specific queue instead of the shared queue. This functionality is covered by the `HardcodeReplyToAddressToLogicalAddress` behavior.
+
+
+## Receiver Side Distribution
+
+Receiver Side Distribution validates if a replica of a partitioned endpoint can process an incoming message or should forward it to the appropriate replica instead. Partition validation is performed by inspecting message headers and the message body.
```

Comment:
This is needed because:
 1. Senders might have no partition aware routing configured.
 2. It is not possible to provide such configuration e.g. in case of native pub-sub sender cannot add this information as it does not know the receivers

This is obligatory thing, always needed and should be configured always for partitioned endpoint as it ensures safety. The client side routing configuration is optional and can be seen an optimization.